### PR TITLE
Added Linter, Prettier, Husky config files and Challenge files

### PR DESCRIPTION
**Pull request by harsh-caw**

### CODE REVIEW

File: Challenge4/app.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    for(let i=0; i<keys.length; i++){
+        let key = keys[i];
+        if(key.className == "key jiggle" && keyPressed.toUpperCase() == key.dataset.key.toUpperCase()){
+            key.className = "key";
+            keys[generateRandomIndex()].className = "key jiggle";
+        }
+    }
+}, false);
+
+function generateRandomIndex(){
```
Comment (vrknetha) [inline Challenge4/app.js]: convert it to arrow function

File: Challenge5/app.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    'name': 'Starting your own Podcast'
+  },
+  {
+    'id': 26,
+    'name': 'How Blogging Can Change Your Career'
+  },
+  {
+    'id': 27,
+    'name': 'Talking to Some of Our Favorite Content Creators'
+  },
```
Comment (vrknetha) [inline Challenge5/app.js]: move it to an other file

File: Challenge1/index.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+}
+
+function updateCountdown(){
+    let minutes = Math.floor(time/60);
+    let seconds = time%60;
+    perc = Math.ceil(((totalsecs - time) / totalsecs) * 100);
+    setProgress(perc);
+    minutes = minutes < 10 ? '0'+minutes : minutes;
+    seconds = seconds < 10 ? '0'+seconds : seconds;
```
Comment (vrknetha) [inline Challenge1/index.js]: convert all to arrow functions and also move other functions to utilities.js and import from there

File: Challenge2/app.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            </button>`;
+    let inCartButton = document.createElement("in-cart");
+    inCartButton.innerHTML = inCartButtonContents;
+    button.parentNode.replaceChild(inCartButton, button);
+
+    updateCartTotal()
+}
+
+function addItemToCart(title, price, imageSrc, id) {
+    var cartRow = document.createElement('li')
```
Comment (vrknetha) [inline Challenge2/app.js]: do not use var, onlyuse let or const and a single file should not have more lines of code. move the functions to an other file and call it from there

File: .eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+      "comma-spacing": "off",
+      "comma-style": "off",
+      "computed-property-spacing": "off",
+      "consistent-this": "off",
+      "eol-last": "off",
+      "func-call-spacing": "off",
+      "func-name-matching": "off",
+      "func-names": "off",
+      "func-style": "off",
+      "function-call-argument-newline": "off",
```
Comment (vrknetha) [inline .eslintrc.json]: you have turned off all the rules. what is the use of having it?

**Review by vrknetha — CHANGES_REQUESTED** [review]

File: Challenge5/app.js (hunk 2 above)
Comment (harsh-caw) [reply Challenge5/app.js]: moved it to an other file

File: Challenge4/app.js (hunk 1 above)
Comment (harsh-caw) [reply Challenge4/app.js]: done

File: Challenge1/index.js (hunk 3 above)
Comment (harsh-caw) [reply Challenge1/index.js]: done

File: .eslintrc.json (hunk 5 above)
Comment (harsh-caw) [reply .eslintrc.json]: changed rules

**Pull request closed without merging**